src/Apps/Order/Utils/useOrderTracking.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  ActionType,
+  ClickedAddNewShippingAddress,
+  ClickedSelectShippingOption,
+  ClickedShippingAddress,
+  ContextModule,
+  ErrorMessageViewed,
+  OwnerType,
+} from "@artsy/cohesion"
+import { useAnalyticsContext } from "System/Analytics/AnalyticsContext"
+import { useMemo } from "react"
+import * as DeprecatedSchema from "@artsy/cohesion/dist/DeprecatedSchema"
+import { useTracking } from "react-tracking"
+import { FulfillmentType } from "Apps/Order/Routes/Shipping2/shippingUtils"
+
+export const useOrderTracking = () => {
+  const { trackEvent } = useTracking()
+  const analytics = useAnalyticsContext()
+  const contextPageOwnerId = analytics.contextPageOwnerId!
+
+  const trackingCalls = useMemo(() => {
+    return {
+      clickedShippingAddress: () => {
+        const payload: ClickedShippingAddress = {
+          action: ActionType.clickedShippingAddress,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: "orders-shipping",
+          context_page_owner_id: contextPageOwnerId,
+        }
+
+        trackEvent(payload)
+      },
+
+      clickedSelectShippingOption: (newShippingQuoteId: string) => {
+        const payload: ClickedSelectShippingOption = {
+          action: ActionType.clickedSelectShippingOption,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: "orders-shipping",
+          subject: newShippingQuoteId,
+          context_page_owner_id: contextPageOwnerId,
+        }
+
+        trackEvent(payload)
+      },
+
+      errorMessageViewed: ({
```

Comment:
**nitpick:** `displayedErrorMessage` may be a better name to match the other methods in this hook.

Reply:
If we are not destructuring and saying something like `orderTracking.errorMessageViewed()` then this makes sense to me for nice intuitive mapping. No strong opinions though.